### PR TITLE
downgrade langchain to < 0.2.0 in nlp recipes

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
Addresses: https://redhat-internal.slack.com/archives/C06S75ZF9JT/p1716305405907729?thread_ts=1716299527.009829&cid=C06S75ZF9JT
/cc @MichaelClifford @jeffmaury

### CODE REVIEW

File: recipes/natural_language_processing/codegen/app/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 langchain_openai
-langchain
+langchain<0.2.0
 streamlit
```
Comment (lmilbaum) [inline recipes/natural_language_processing/codegen/app/requirements.txt]: Can you please add a newline?

File: recipes/natural_language_processing/codegen/app/requirements.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
 langchain_openai
-langchain
+langchain<0.2.0
```
Comment (lmilbaum) [inline recipes/natural_language_processing/codegen/app/requirements.txt]: I would suggest to pin to a specific version. Is that possible?

File: recipes/natural_language_processing/chatbot/app/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
-langchain
+langchain<0.2.0
 langchain_openai
 streamlit
```
Comment (lmilbaum) [inline recipes/natural_language_processing/chatbot/app/requirements.txt]: Please add a newline.

**Review by lmilbaum — CHANGES_REQUESTED** [review]

File: recipes/natural_language_processing/codegen/app/requirements.txt [hunk 4]
```diff
@@ -1,3 +1,3 @@
 langchain_openai
```
Comment (jeffmaury) [inline recipes/natural_language_processing/codegen/app/requirements.txt]: Shouldn't we restrict this as well ?

Comment (Gregory-Pereira) [reply recipes/natural_language_processing/codegen/app/requirements.txt]: I have since locked all of them

File: renovate.json [hunk 5]
```diff
@@ -2,5 +2,11 @@
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
   "extends": [
     "github>platform-engineering-org/.github"
+  ],
+  "packageRules": [
+    {
+      "matchDatasources": ["pypi"],
```
Comment (lmilbaum) [inline renovate.json]: Hmmm... not sure if that is needed. What is needed is to pin the version of one dependency.

**Review by lmilbaum — CHANGES_REQUESTED** [review]

**Review by lmilbaum — APPROVED** [review]
LGTM

File: renovate.json [hunk 6]
```diff
@@ -2,5 +2,15 @@
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
   "extends": [
     "github>platform-engineering-org/.github"
+  ],
+  "packageRules": [
+    {
+      "matchPackageNames": ["langchain"],
+      "allowedVersions": "<=0.2.0"
```
Comment (jeffmaury) [inline renovate.json]: 0.2.0 should be excluded

**Pull request merged**